### PR TITLE
K8SPXC-523 Fix annotations merge

### PR DESCRIPTION
**Pull request by nonemax**
[K8SPXC-523](https://jira.percona.com/browse/K8SPXC-523)

### CODE REVIEW

File: pkg/controller/pxc/upgrade.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func (r *ReconcilePerconaXtraDBCluster) updatePod(sfs api.StatefulApp, podSpec *
 	currentSet.Spec.UpdateStrategy = sfs.UpdateStrategy(cr)
 
 	// support annotation adjustements
-	currentSet.Spec.Template.Annotations = podSpec.Annotations
+	pxc.MergeTmplateAnnotations(currentSet, podSpec.Annotations)
```
Comment (cap1984) [inline pkg/controller/pxc/upgrade.go:38]: typo

Comment (nonemax) [reply pkg/controller/pxc/upgrade.go]: fixed https://github.com/percona/percona-xtradb-cluster-operator/pull/668/commits/59292fae2e82ae5cef4df82763e5e84267b30295

**Review by cap1984 — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | failed |
| storage | passed |
| big-data | passed |
| recreate | passed |
| scaling | passed |
| haproxy | passed |
| self-healing | passed |
| init-deploy | failed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| security-context | failed |
| self-healing-advanced | passed |
| demand-backup | passed |
| operator-self-healing | failed |
| limits | failed |
| monitoring | passed |
| monitoring-2-0 | passed |
| scheduled-backup | passed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| upgrade-proxysql | failed |
| smart-update | failed |
| upgrade-consistency | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scaling | passed |
| recreate | passed |
| haproxy | passed |
| storage | passed |
| init-deploy | failed |
| scaling-proxysql | failed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| self-healing-advanced | passed |
| operator-self-healing | failed |
| demand-backup | passed |
| limits | failed |
| monitoring | passed |
| monitoring-2-0 | passed |
| scheduled-backup | passed |
| affinity | passed |
| one-pod | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

**Review by tplavcic — APPROVED** [review]

**Pull request merged**